### PR TITLE
Use std::sort in suitable places

### PR DESCRIPTION
**Pull request by data-man**

### CODE REVIEW

File: slob.cc [hunk 1]
```diff
@@ -414,7 +415,7 @@ const SlobFile::RefOffsetsVector & SlobFile::getSortedRefOffsets()
       refsOffsetVector.append( RefEntryOffsetItem( base + qFromBigEndian( tmp ), i ) );
     }
 
-    qSort( refsOffsetVector );
+    std::sort( refsOffsetVector );
```
Comment (vedgy) [inline slob.cc:418]: Fixing deprecation warnings is nice. But did you try compiling this? `std::sort` requires at least two arguments. The same issue in other places.

File: favoritespanewidget.cc [hunk 2]
```diff
@@ -906,7 +907,7 @@ void FavoritesModel::removeItemsForIndexes( const QModelIndexList & idxList )
   for( int i = lowestLevel; i >= 0; i-- )
   {
     QModelIndexList idxSublist = itemsToDelete[ i ];
-    qSort( idxSublist.begin(), idxSublist.end(), qGreater< QModelIndex >() );
+    std::sort( idxSublist.begin(), idxSublist.end(), qGreater< QModelIndex >() );
```
Comment (vedgy) [inline favoritespanewidget.cc:910]: `qGreater` is deprecated too. Might as well fix it too along with another usage below.

File: slob.cc (hunk 1 above)
Comment (data-man) [reply slob.cc:418]: > But did you try compiling this?

OMG, I didn't notice any compilation errors and ran the old binary. :(
I'll fix it soon™.

**Comment by vedgy** [comment]
Note: #1246 fixed more deprecation warnings and changed the configuration directory location. It was closed because the configuration directory change was questionable and many changes didn't compile with Qt4.

File: zim.cc [hunk 3]
```diff
@@ -42,6 +42,8 @@
 #include <string>
 #include <set>
 #include <map>
+#include <algorithm>
+#include <functional>
```
Comment (vedgy) [inline zim.cc:46]: `<functional>` should be included only in the files where you use `std::greater`.

File: favoritespanewidget.cc [hunk 4]
```diff
@@ -906,7 +907,7 @@ void FavoritesModel::removeItemsForIndexes( const QModelIndexList & idxList )
   for( int i = lowestLevel; i >= 0; i-- )
   {
     QModelIndexList idxSublist = itemsToDelete[ i ];
-    qSort( idxSublist.begin(), idxSublist.end(), qGreater< QModelIndex >() );
+    std::sort( idxSublist.begin(), idxSublist.end() );
```
Comment (vedgy) [inline favoritespanewidget.cc:910]: Simply removing `qGreater` changes the application behavior - here and below.

File: zim.cc [hunk 5]
```diff
@@ -290,7 +292,7 @@ bool ZimFile::open()
   for( quint32 i = 0; i < zimHeader.clusterCount; i++ )
     clusterOffsets[ i ] = QPair< quint64, quint32 >( offs.at( i ), i );
 
-  qSort( clusterOffsets );
+  std::sort(clusterOffsets.begin(), clusterOffsets.end());
```
Comment (vedgy) [inline zim.cc:295]: Please restore spaces after/before the parentheses here and in another place above.

**Review by vedgy — CHANGES_REQUESTED** [review]

**Comment by vedgy** [comment]
@data-man, can you please review your own diff? There is still a behavior change, a formatting mistake and `qGreater` is not replaced with `std::greater` in the modified lines.

**Comment by data-man** [comment]
@vedgy 

```c++
std::greater< QModelIndex >
```
can`t be used because:

> /usr/include/c++/10.2.0/bits/stl_function.h:376:20: error: no match for ‘operator>’ (operand types are ‘const QModelIndex’ and ‘const QModelIndex’)

**Comment by vedgy** [comment]
> can`t be used

All right, but the other two issues remain.

File: zim.cc (hunk 5 above)
Comment (data-man) [reply zim.cc]: `std::sort( clusterOffsets.begin(), clusterOffsets.end() );`
What's wrong here?

**Comment by data-man** [comment]
> There is still a behavior change

Why?

File: historypanewidget.cc [hunk 6]
```diff
@@ -137,7 +138,7 @@ void HistoryPaneWidget::deleteSelectedItems()
 
   // Need to sort indexes in the decreasing order so that
   // the first deletions won't affect the indexes for subsequent deletions.
-  qSort( idxsToDelete.begin(), idxsToDelete.end(), qGreater<int>() );
+  std::sort( idxsToDelete.begin(), idxsToDelete.end() );
```
Comment (vedgy) [inline historypanewidget.cc:141]: The behavior change is here. `#include <functional>` and use `std::greater<int>`.

File: fulltextsearch.cc [hunk 7]
```diff
@@ -665,7 +665,7 @@ Q_UNUSED( parent );
   }
 
   headwords.append( temp );
-  qSort( headwords );
+  std::sort( headwords.begin(),  headwords.end());
```
Comment (vedgy) [inline fulltextsearch.cc:668]: The formatting mistake is here.

**Review by vedgy — CHANGES_REQUESTED** [review]

File: zim.cc (hunk 5 above)
Comment (vedgy) [reply zim.cc]: My comment ends with *and in another place above.*

File: fulltextsearch.cc (hunk 7 above)
Comment (data-man) [reply fulltextsearch.cc]: Thanks!

File: historypanewidget.cc [hunk 8]
```diff
@@ -137,7 +138,7 @@ void HistoryPaneWidget::deleteSelectedItems()
 
   // Need to sort indexes in the decreasing order so that
   // the first deletions won't affect the indexes for subsequent deletions.
-  qSort( idxsToDelete.begin(), idxsToDelete.end(), qGreater<int>() );
+  std::sort( idxsToDelete.begin(), idxsToDelete.end(), std::greater<int>() );
```
Comment (vedgy) [inline historypanewidget.cc:141]: Include what you use. `std::greater` is not guaranteed to compile on every supported platform without `<functional>`.

**Review by vedgy — CHANGES_REQUESTED** [review]
Please squash all commits into one and force-push. These mistakes won't be interesting to future developers.

**Review by vedgy — APPROVED** [review]
Looks good now.

**Comment by vedgy** [comment]
@Abs62, why not merge this pull request? `qSort` and `qGreater` have been deprecated for many years in Qt 5 and cause compiler warnings. They were removed in Qt 6.

**Pull request merged**